Add Vanilla Fullpage Scroll Solution (Alternative to Fullpage.js)

samuelsogbesan opened the pull request:
- **Add bare-bones vanilla view-port scroll functionality in CSS (Previously managed by Fullpage.js)**
**_INFORMATION AROUND DECISION_**: Although applicable, I have decided to no longer depend on Fullpage.js since I have been able to implement the core functionality, the full page scroll, in vanilla CSS. A decision in the future may be made to incorporate Fullpage.js for its animation and navigation capabilities, hence why the structure of the static content mirrors the required structure for Fullpage.js.

The CSS file describes a wrapper (#fullscreen) that contains sections (.section) that occupy the full size of the viewport. 

- **Adjust structure of HTML document to support new styling whilst being compatible with Fullpage.js (In the case the license is cleared and I revert).**
The HTML document structure was updated to follow the structure required for Fullpage.js support here: [https://github.com/alvarotrigo/fullPage.js#required-html-structure](https://github.com/alvarotrigo/fullPage.js#required-html-structure)
Although the project doesn't rely on the framework, I decided to make the structure compatible in case some future functionality from it is incorporated.

samuelsogbesan committed:
Add vanilla full-viewport styling for fullscreen wrapper and sections

samuelsogbesan committed:
Adjust HTML structure for vanilla solution (also compatible with fullpage.js)

samuelsogbesan committed:
Remove Section boilerplate information

samuelsogbesan committed:
Add remaining sections for each part of portfolio

krasimirgg commented:
Consider updating the PR comment: https://github.com/samuelsogbesan/step/pull/4#issue-449441748 with an up-to-date information about your decision to use or not the fullpage library. In particular, the '(Previously managed by Fullpage.js)' bit.

krasimirgg commented on portfolio/src/main/webapp/index.html:
```diff
@@ -1,16 +1,26 @@
-</html>
+
+<head>
+	<meta charset="UTF-8">
```
Please use indentation of +2 spaces (instead of the current +4), per our style guide:
https://google.github.io/styleguide/htmlcssguide.html#Indentation

samuelsogbesan replied:
Managed to replace the tab indentation with 2 space. See commit 9af3ca3.

krasimirgg commented on portfolio/src/main/webapp/index.html:
```diff
@@ -1,16 +1,26 @@
+</head>
+
+<body>
+	<div id="fullpage">
```
This PR appears to be doing two independent things: removing the old HTML content *and* adding fullpage functionality. Consider splitting those up or applying just one of those things. Also, I suggest you don't leave all those empty sections -- think about what you wanna put in there; maybe use some placeholders, or consider adding [TODOs](https://google.github.io/styleguide/htmlcssguide.html#Action_Items).

samuelsogbesan replied:
Thanks, I'll push a commit through soon that adds some TODOs.

samuelsogbesan replied:
I'll also consider how to split up the PR.

samuelsogbesan replied:
@krasimir-google, I've gone ahead and reverted the commit 5e68066 and have added back the boilerplate content. I will create a separate pull request to remove this content.

krasimirgg commented on portfolio/src/main/webapp/index.html:
```diff
@@ -1,16 +1,26 @@
+		</div>
+		<div class="section">
+		</div>
+        <div class="section">
```
The indentation of these is misleading -- it suggests that maybe the first 2 `<div>`-s are somehow at a different level in the DOM three than the later 3.

samuelsogbesan replied:
Indentation resolved with later commits 3d52c29c49bc4bb0319343aece1bae300b61a104, 9af3ca31c76c09be4cdd03c7bc2c351703e166f1, 3e3a7fb48fd3ebdceb9da199b8ca059eed448a23

krasimirgg commented on portfolio/src/main/webapp/style.css:
```diff
@@ -0,0 +1,21 @@
+body{
```
Use an indentation of 2 spaces for CSS and add a space between a selector and the opening `{`, as you can see in our style guide examples:
https://google.github.io/styleguide/htmlcssguide.html#Indentation

samuelsogbesan replied:
Thanks, I'll go ahead and push through a commit with the required spacing and a separate commit for the Indentation.

samuelsogbesan commented:
> Consider updating the PR comment: [#4 (comment)](https://github.com/samuelsogbesan/step/pull/4#issue-449441748) with an up-to-date information about your decision to use or not the fullpage library. In particular, the '(Previously managed by Fullpage.js)' bit.

@krasimir-google, Thanks, I've gone ahead and updated the comment with information on my decision.

samuelsogbesan committed:
Add TO-DOs to empty sections

samuelsogbesan committed:
Add space between selector and opening bracket as specified by style guide

samuelsogbesan committed:
Change tab spacing from +4 to +2 on index.html

samuelsogbesan committed:
Replace Tabs with +2 Spaces

Used the tool https://tabstospaces.com/ to automatically convert my tabs to 2 spaces!

samuelsogbesan committed:
Change from tab indentation to +2 Spaces

samuelsogbesan committed:
Revert @5e68066 (Add boilerplate content back)

samuelsogbesan committed:
Update indentation to align with +2 in index.html

krasimirgg commented on portfolio/src/main/webapp/index.html:
```diff
@@ -1,16 +1,34 @@
       <h1>My Portfolio</h1>
       <p>This is my portfolio.</p>
-      <p>Click here to get a random message:</p>
+      <p>Click here to get a random message:</p>      
```
Please remove the trailing whitespace:
https://google.github.io/styleguide/htmlcssguide.html#Trailing_Whitespace

samuelsogbesan replied:
Trailing Whitespace removed in commit 971a8c4

krasimirgg commented on portfolio/src/main/webapp/style.css:
```diff
@@ -0,0 +1,20 @@
+}
+
+.section {
+  min-height: 100vh;
```
Please list CSS properties in alphabetic order:
https://google.github.io/styleguide/htmlcssguide.html#Declaration_Order
(here and in other groupsets)

samuelsogbesan replied:
Corrected in commit 8eecc65

krasimirgg replied:
I believe this CSS property section is still not sorted alphabetically, e.g., `max-height` should precede`min-height`; `overflow` should be last, etc.

samuelsogbesan replied:
I've gone ahead and re-attempted the alphabetization with a [tool](https://www.freeformatter.com/css-beautifier.html) and have pushed this with commit e0eff1e

krasimirgg commented on portfolio/src/main/webapp/style.css:
```diff
@@ -0,0 +1,20 @@
+body {
+  margin:0;
```
please add a space after `:`

samuelsogbesan replied:
Thanks, corrected in commit bdd8f61

samuelsogbesan commented:
Re-Review Requested by accident on this PR, changes to be made!

samuelsogbesan committed:
Remove trailing whitespace

samuelsogbesan committed:
Remove duplicate declaration for #fullpage rule

samuelsogbesan committed:
Remove trailing whitespace on CSS file

samuelsogbesan committed:
Alphabetise declaration order for #fullpage

samuelsogbesan committed:
Add space betweeen colon and value

samuelsogbesan committed:
Add centering to section styling and alphabetise its attributes

krasimirgg reviewed (approved):
Thanks!
LGTM!
